src/App.tsx

```diff
@@ -1,60 +1,123 @@
+  ...todo,
+  user: getUserByID(todo.userId),
+}));
 
 export const App = () => {
+  const [preparedTodo, setPreparedTodo] = useState(todos);
+  const [TodoTitle, setTodoTitle] = useState('');
+  const [isPressed, setIsPressed] = useState(false);
+  const [UserId, setUserId] = useState(0);
+
+  const userRuleErr = isPressed && UserId === 0;
+  const titleRuleErr = isPressed && TodoTitle.length === 0;
+
+  const handleAddTodo = () => {
+    const id = preparedTodo.map(todo => todo.id);
+
+    if (TodoTitle.length > 0 && TodoTitle.trim() && UserId !== 0) {
+      const newTodo: Todo = {
+        id: Math.max(...id) + 1,
+        title: TodoTitle,
+        userId: UserId,
+        completed: false,
+        user: getUserByID(UserId),
+      };
+
+      setPreparedTodo(state => [...state, newTodo]);
+      setTodoTitle('');
+      setUserId(0);
+    }
+  };
+
+  const addingAndCheckingTodo = () => {
+    handleAddTodo();
+
+    if (TodoTitle.length !== 0 && UserId !== 0) {
+      setIsPressed(false);
+    }
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTodoTitle(event.target.value);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(Number(event.target.value));
+  };
+
+  const handleButtonSubmit = (event:
+  React.MouseEvent<HTMLButtonElement,
+  MouseEvent>) => {
+    event.preventDefault();
+    setIsPressed(true);
+    addingAndCheckingTodo();
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
       <form action="/api/users" method="POST">
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <input
+            type="text"
+            data-cy="titleInput"
+            value={TodoTitle}
+            onChange={handleTitleChange}
+            placeholder="Enter a title"
+            required
+          />
+          {titleRuleErr && <span className="error">Please enter a title</span>}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
+          <select
+            value={UserId}
+            data-cy="userSelect"
+            onChange={handleUserChange}
+          >
+            <option disabled value="0">Choose a user</option>
+
+            {usersFromServer.map(user => (
+              <option
+                key={user.id}
+                value={user.id}
+              >
+                {user.name}
+              </option>
+            ))}
           </select>
 
-          <span className="error">Please choose a user</span>
+          {userRuleErr && <span className="error">Please choose a user</span>}
         </div>
 
-        <button type="submit" data-cy="submitButton">
+        <button
+          type="submit"
+          data-cy="submitButton"
+          onClick={handleButtonSubmit}
```

Comment:
Event handler should be added to form. Check Misha's video on sign up form.
https://mate.academy/learn/react/react-forms#/video/369 and cheatsheet: https://mate-academy.github.io/fe-program/js/extra/react-typescript#forms-events